### DISCUSSION TITLE
--cert option with password -> file not found

### DISCUSSION THREAD

**Discussion by mm202** [Orange-OpenSource/hurl#2013] [Q&A] [answered]
Hi,

I am a new hurl user and therefore, not quite sure whether my problem is an issue or just me beeing stupid^^

I would like to make a POST request and use a client certificate (.pfx) that is encrypted with a password. As far as I understand the documentation, this should work with:

`hurl --cert ./certificate.pfx:password` 

Unfortunately, this results in an error message: "error: no such file". It seems to me, hurl interprets the password as part of the file path.

After splitting the .pfx file in a cert.crt and a key.key file without a password, the request works as expected.

OS: macos Sonoma
Terminal: iterm2
hurl: installed via homebrew a few days ago

Is this a bug or is there anything I am doing wrong?

Thank you!

Ragards
Martin

**Comment by jcamiel** [comment]
Hi @mm202 you're not stupid, it's probably a bug with Hurl😓!

I wasn't aware of the possibility to provide a password with `--cert` option and looking at https://github.com/Orange-OpenSource/hurl/blob/af3f78e637cbed599431c09d5fea707349a23453/packages/hurl/src/http/client.rs#L149-L151 I've the impression that we don't manage the optional password and use the cert option value as a file. Just need to check it with @fabricereix and we'll probably convert this discussion to a bug report.

Thank for this!

**Reply by mm202** [reply]
Hi,

Is there any update on this? Should I create a bug report?

Thanks and regards,
Martin

**Comment by jcamiel** [comment]
Hi @mm202 sorry for this, I've forgot to convert this discussion to an issue!

**Comment by jcamiel** [comment, marked as answer]
Hi @mm202 this has been implemented in #2246 and will be release in Hurl 4.2.0
